ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/131__rhods_operatorv2.robot

```diff
@@ -0,0 +1,180 @@
+    IF    "${component}" not in @{COMPONENTS} and "${component}" != "none"
+        Log    unknown component: ${component}
+        RETURN
+    END
+    ${len} =    Get Length    ${COMPONENTS}
+    ${patch} =    Set Variable    ${PATCH_PREFIX}
+    FOR    ${index}    ${cmp}    IN ENUMERATE    @{COMPONENTS}
+        IF     "${cmp}"=="${component}"
+            ${sub} =    Change Component Status    component=${cmp}    enable=true    run=${FALSE}
+        ELSE
+            ${sub} =    Change Component Status    component=${cmp}    enable=false    run=${FALSE}
+        END
+        IF    ${index} ==0
+            ${patch} =    Catenate    SEPARATOR=    ${patch}    ${sub}
+        ELSE
+            ${patch} =    Catenate    SEPARATOR=,${SPACE}    ${patch}    ${sub}
+        END
+    END
+    ${patch} =    Catenate    SEPARATOR=    ${patch}    }}}'
+    Log    ${patch}
+    ${status} =    Run    ${patch}
+    Log    ${status}
+    Sleep    30s
+
+Change Component Status
+    [Documentation]    Enables or disables a single component. Can either run the patch command directly (if `run` is
+    ...    set to true) or return the patch string to be combined later for a bigger patch command.
+    [Arguments]    ${component}    ${run}=${TRUE}    ${enable}=true
+    IF    "${component}" not in @{COMPONENTS}
+        Log    unknown component: ${component}
+        RETURN
+    END
+    IF    ${run}==${TRUE}
+        ${command} =    Catenate    SEPARATOR=    ${PATCH_PREFIX}   "${component}":{"enabled": ${enable}}    }}}'
+        ${status} =    Run    ${command}
+        Log    ${status}
+        Sleep    30s
+    ELSE
+        RETURN    "${component}":{"enabled": ${enable}}
+    END
+
+Get Original Configuration
+    [Documentation]
+    @{config} =    Create List
+    FOR    ${cmp}    IN    @{COMPONENTS}
+        ${status} =    Is Component Enabled    ${cmp}
+        Append To List    ${config}    ${status}
+    END
+    ${ORIGINAL_CONFIGURATION} =    ${config}
+
+Patch DataScienceCluster CustomResource To Original Configuration
+    [Documentation]  Enables a mix of components based on the values set before this test suite was started
+    [Arguments]    ${dsc_name}=default
+    ${len} =    Get Length    ${COMPONENTS}
+    ${patch} =    Set Variable    ${PATCH_PREFIX}
+    FOR    ${index}    ${cmp}    IN ENUMERATE    @{COMPONENTS}
+        IF    "${ORIGINAL_CONFIGURATION}[${index}]" == "true"
+            ${sub} =    Change Component Status    component=${cmp}    run=${FALSE}    enable=true
+        ELSE
+            ${sub} =    Change Component Status    component=${cmp}    run=${FALSE}    enable=false
+        END
+        IF    ${index} ==0
+            ${patch} =    Catenate    SEPARATOR=    ${patch}    ${sub}
+        ELSE
+            ${patch} =    Catenate    SEPARATOR=,${SPACE}    ${patch}    ${sub}
+        END
+    END
+    ${patch} =    Catenate    SEPARATOR=    ${patch}    }}}'
+    Log    ${patch}
+    ${status} =    Run    ${patch}
```

Comment:
should we check and fail in case the run fails?

Reply:
it might be good to fail as soon as possible instead of going forward, but it depends on whats next in the test. Anyway, I don't think it's critical now, feel free to ignore

Reply:
Added this check